sotodlib/tod_ops/flags.py

```diff
@@ -672,3 +672,72 @@ def get_inv_var_flags(aman, signal_name='signal', nsigma=5,
 
     return mskinvar
+
+def get_split_flags(aman, proc_aman=None, split_cfg=None):
+    '''
+    Function returning flags used for null splits consumed by the mapmaking and bundling codes.
+    Fields labeled ``field_name_flag`` contain boolean masks and ``_avg`` are the mean
+    of the numerical based split flags to be used for observation level splits.
+
+    Arguments
+    ---------
+    aman: AxisManager
+        Main axis manager containing signal.
+    proc_aman: AxisManager
+        Preprocess axis manager, usually loaded in ``aman.preprocess``.
+    split_cfg: dict
+        Dictionary containing the thresholds used for cutting
+
+    Returns
+    -------
+    split_aman: AxisManager
+        Axis manager containing splitting flags.
+    '''
+    if proc_aman is None:
+        try:
+            proc_aman = aman.preprocess
+        except:
+            raise ValueError('proc_aman is None and no preprocess field in aman provide valid preprocess metadata')
+
+    # Set default set of splits
+    default_cfg = {'high_gain': 0.115, 'high_noise': 3.5e-5, 'high_tau': 1.5e-3, 
+                   'det_A': 'A', 'pol_angle': 35, 'det_top': 'B', 'high_leakage': 1e-3,
+                   'high_2f': 1.5e-3, 'right_focal_plane': 0, 'top_focal_plane': 0,
+                   'central_pixels': 0.071 }
+    if split_cfg is None:
+        split_cfg = default_cfg
+
+    split_aman = core.AxisManager(aman.dets)
+    # If provided split config doesn't include all of the splits in default 
+    for k in default_cfg.keys():
+        if not k in split_cfg:
+            split_cfg[k] = default_cfg[k]
+        split_aman.wrap(f'{k}_threshold', split_cfg[k])
+
+    split_aman.wrap('high_gain_flag', aman.det_cal.phase_to_pW > split_cfg['high_gain'],
+                    [(0, 'dets')])
+    split_aman.wrap('high_gain_avg', np.nanmean(aman.det_cal.phase_to_pW))
+    split_aman.wrap('high_noise_flag', proc_aman.noiseQ_fit.fit[:,1] > split_cfg['high_noise'],
+                    [(0, 'dets')])
+    split_aman.wrap('high_noise_avg', np.nanmean(proc_aman.noiseQ_fit.fit[:,1]))
+    split_aman.wrap('high_tau_flag', aman.det_cal.tau_eff > split_cfg['high_tau'], 
+                    [(0, 'dets')])
+    split_aman.wrap('high_tau_avg', np.nanmean(aman.det_cal.tau_eff))
+    split_aman.wrap('det_A_flag', aman.det_info.wafer.pol <= split_cfg['det_A'],
+                    [(0, 'dets')])
+    split_aman.wrap('pol_angle_flag', aman.det_info.wafer.angle > split_cfg['pol_angle'], 
+                    [(0, 'dets')])
+    split_aman.wrap('det_top_flag', aman.det_info.wafer.crossover > split_cfg['det_top'], 
+                    [(0, 'dets')])
+    split_aman.wrap('high_leakage_flag', np.sqrt(proc_aman.t2p.lamQ**2 + proc_aman.t2p.lamU**2) > split_cfg['high_leakage'],
+                    [(0, 'dets')])
+    split_aman.wrap('high_leakage_avg', np.nanmean(np.sqrt(proc_aman.t2p.lamQ**2 + proc_aman.t2p.lamU**2)),
```

Comment:
Maybe this should either just be `leakage_average` if it is going to calculate it across all `lamQ` and `lamU` values or should we actually only calculate the average from the "high" values (or have two fields for high and low)?